File: jewels-stones.java
class Solution {
public:
    int numJewelIsStones(string J, string S){
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);

        unordered_map<char, int> mymap;
        for(int i = 0; j[i] != '\0'; i++)
            ++mymap[j[i]];

        int count = 0;
        for(int i=0;S[i]!='\0';++1)
            if(mymap.find(S[i])!=mymap.end())
                count+=1;

        return count;
    }
};

    

    

File: jewels-stones.py
def numJewelIsStones(J: str, S: str) -> int:
    mymap = {}
    for ch in J:
        if ch not in mymap:
            mymap[ch] = 0
        mymap[ch] += 1
    
    count = 0
    for ch in S:
        if ch in mymap:
            count += 1
    
    return count